lib/Runtime/Library/DataView.h

```diff
@@ -55,6 +55,34 @@ namespace Js
             return static_cast<DataView*>(aValue);
         }
 
+        static uint32 ToIndex (Var value, ScriptContext *scriptContext)
```

Comment:
How is it different than ArrayBuffer::ToIndex ?

Reply:
Oh you mentioned that in the description that they are almost similar. Good to have one of them.

---
In reply to: [181444808](https://github.com/Microsoft/ChakraCore/pull/4981#discussion_r181444808) [](ancestors = 181444808)

Reply:
Do you want me to refactor this to share one method or leave as is?

Reply:
If it is straightforward to capture the difference, better to refactor. To avoid compiling in unnecessary branches (since the actual method needed in each case is known to us at compile-time) you can use a `template <bool SomeConditional>` strategy.